api/api/views/report.py

```diff
@@ -0,0 +1,415 @@
+import base64
+import os
+from datetime import datetime
+from typing import Type, List
+import pdfkit
+from django.db.models import CharField
+from django.http import HttpResponseRedirect
+from shutil import rmtree
+
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.status import HTTP_404_NOT_FOUND
+from rest_framework.views import APIView
+
+from api.models import Team, Manager
+from api.models.mission import Mission
+from api.models.vulns import Vulnerability
+from api.services.s3 import S3Bucket
+
+DIR_STYLE = "./api/pdf-templates/hackmanit-template"
+CSS_PATH = f'{DIR_STYLE}/main.css'
+ABSOLUTE_DIR_STYLE = os.path.abspath(DIR_STYLE if os.getenv("PRODUCTION", 0) else DIR_STYLE.replace("api/", ""))
+ABSOLUTE_CSS_PATH = f"{ABSOLUTE_DIR_STYLE}/main.css"
+
+
+def get_image_file_as_base64_data(path):
+    with open(path, 'rb') as image_file:
+        return base64.b64encode(image_file.read())
+
+
+def generate_members(team: Team):
+    members_html = ""
+    for member in team.members.all():
+        members_html += f"<p>{member.auth.first_name} {member.auth.last_name}</p>"
+    return members_html
+
+
+class GenerateReportView(APIView):
+    permission_classes = [AllowAny]
+    authentication_classes: List[Type[TokenAuthentication]] = []
+
+    def get(self, request):
+        mission_id = request.data.get("mission")
+        if not mission_id:
+            return Response({
+                'error': 'No mission id provided. Report couldn\'t be generated',
+            }, status=HTTP_404_NOT_FOUND)
+        mission = Mission.objects.filter(pk=mission_id).first()
+        if not mission:
+            return Response({
+                'error': f'No mission with id {mission_id}. Report couldn\'t be generated',
+            }, status=HTTP_404_NOT_FOUND)
+        s3 = S3Bucket()
+        s3.create_bucket(mission.bucket_name)
+        dir_path = f'/tmp/{mission.bucket_name}'
+        filepath = self.dump_report(mission, dir_path)
+        object_name = f'report-{mission.title}.pdf'
```

Comment:
this is probably vulnerable to LFI, maybe you should generate a random name instead